File: Components/Parameter/ParameterSettingsControlBase.vb
'***************************************************************************/
'* ParameterSettingsBase.vb
'*
'* COPYRIGHT (c) 2004-2005 by DNNStuff
'* ALL RIGHTS RESERVED.
'*
'* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED 
'* TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL 
'* THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
'* CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER 
'* DEALINGS IN THE SOFTWARE.
'*************/
Namespace DNNStuff.SQLViewPro.Controls
    Public MustInherit Class ParameterSettingsControlBase
        Inherits System.Web.UI.UserControl

#Region "Abstract Methods"
        Public MustOverride Sub LoadSettings(ByVal settings As String)
        Public MustOverride Function UpdateSettings() As String
        Protected MustOverride ReadOnly Property LocalResourceFile() As String
        Public Overridable ReadOnly Property CaptionRequired() As Boolean
            Get
                Return True
            End Get
        End Property
#End Region

        Private Sub Page_Load(sender As Object, e As System.EventArgs) Handles Me.Load
            LoadLabelResources(Me)
        End Sub

        Private Sub LoadLabelResources(root As Control)
            For Each c As Control In root.Controls
                If TypeOf (c) Is DotNetNuke.UI.UserControls.LabelControl Then
                    Dim label As DotNetNuke.UI.UserControls.LabelControl = DirectCast(c, DotNetNuke.UI.UserControls.LabelControl)
                    Dim labelText As String = DotNetNuke.Services.Localization.Localization.GetString(label.ID & ".Text", LocalResourceFile)
                    If labelText Is Nothing Then labelText = label.ID.Replace("lbl", "")
                    Dim helpText As String = DotNetNuke.Services.Localization.Localization.GetString(label.ID & ".Help", LocalResourceFile)
                    If helpText Is Nothing Then helpText = "Help not available for " & labelText
                    label.Text = labelText
                    label.HelpText = helpText
                End If
                If c.HasControls() Then
                    LoadLabelResources(c)
                End If
            Next
        End Sub
    End Class
End Namespace

File: Components/Parameter/ParameterSettingsControlBase.cs
using System.Web.UI;

namespace DNNStuff.SQLViewPro.Controls
{
	public abstract class ParameterSettingsControlBase : UserControl
	{
		
#region Abstract Methods
		public abstract void LoadSettings(string settings);
		public abstract string UpdateSettings();
		protected abstract string LocalResourceFile {get;}
		public virtual bool CaptionRequired => true;

	    #endregion
		
		protected void Page_Load(object sender, System.EventArgs e)
		{
			LoadLabelResources(this);
		}
		
		private void LoadLabelResources(Control root)
		{
			foreach (Control c in root.Controls)
			{
				if ((c) is DotNetNuke.UI.UserControls.LabelControl)
				{
					var label = (DotNetNuke.UI.UserControls.LabelControl) c;
					var labelText = (string) (DotNetNuke.Services.Localization.Localization.GetString(label.ID + ".Text", LocalResourceFile));
					if (labelText == null)
					{
						labelText = label.ID.Replace("lbl", "");
					}
					var helpText = (string) (DotNetNuke.Services.Localization.Localization.GetString(label.ID + ".Help", LocalResourceFile));
					if (helpText == null)
					{
						helpText = "Help not available for " + labelText;
					}
					label.Text = labelText;
					label.HelpText = helpText;
				}
				if (c.HasControls())
				{
					LoadLabelResources(c);
				}
			}
		}
	}
}
